src/plots/cartesian/layout_defaults.js

```diff
@@ -55,22 +63,47 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
         }
     }
 
-    function axSort(a,b) {
-        var aNum = Number(a.substr(5)||1),
-            bNum = Number(b.substr(5)||1);
-        return aNum - bNum;
+    // N.B. Ignore orphan axes (i.e. axes that have no data attached to them)
+    // if gl3d or geo is present on graph. This is retain backward compatible.
+    //
+    // TODO drop this in version 2.0
+    var ignoreOrphan = (layoutOut._hasGL3D || layoutOut._hasGeo);
```

Comment:
This is presumably just an issue just for plots made and stored in the workspace right? Can we confirm that we're not generating this situation anymore? (Or perhaps we *are* until we properly handle type changes in `restyle`) And then how hard would it be to write a command to find and clear these orphans out of stored plots, so we never have to put this logic in? Otherwise I don't see how we're going to be able to take it out later.

I suppose we can put this logic in now, deploy it, then at some later time make and run a cleaning command, then come in after THAT and delete this logic... I just worry that if we don't do it now it will become harder later, and therefore will never happen, and this quirk will live forever.